p2panda-stream/src/ordering/partial/store.rs

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::collections::{HashMap, HashSet};
+use std::fmt::Debug;
+use std::hash::Hash as StdHash;
+
+use super::PartialOrderError;
+
+/// Trait defining a store API for handling ready and pending dependencies.
+///
+/// An implementation of this store trait provides the following functionality:
+/// - maintain a list of all items which have all their dependencies met
```

Comment:
Is it possible to only keep track of "pending" items and as soon as they are _not_ in the list anymore they are considered "ready"? That would allow to not grow the size of storage, though it would also not be much more than the operation itself already.

Reply:
I considered this approach, as we discussed it before, however I can't see a way to make it work. If the "pending" items set is empty and we receive a new item with dependencies `[x, y]` we wouldn't know if these were items which have been processed and ready (therefore removed from the pending set), or that we have never seen (and therefore never added to the pending set).

Reply:
Maybe I'm misunderstanding your suggestion though!

Reply:
Ah, yes! Let's not do it then!